### ISSUE TITLE
Document that Maven properties can be used as source for build-time config

### ISSUE THREAD

**Issue by knutwannheden** [quarkusio/quarkus#16283]
## Description

On [Zulip there was a discussion](https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/Why.20are.20there.20no.20maven-plugins.3F) regarding how to tag the Docker images produced by Quarkus using `git rev-parse --short HEAD`. It was pointed out that Maven properties are also a config source for Quarkus' build-time properties. I couldn't find anything regarding this in the documentation:

* https://quarkus.io/guides/maven-tooling
* https://quarkus.io/guides/config

I think it would make sense to add this to one of the mentioned guides and possibly also to https://quarkus.io/guides/container-image, since I imagine it could be a common use case for this particular feature.

**Comment by knutwannheden** [comment]
As commented in the chat I find the precedence ordering somewhat problematic, because overriding an `application.properties` config via `mvn`s `-D` works, but via user property in the `pom.xml` file doesn't. That should probably also be clarified.

**Comment by radcortez** [comment]
Yes, because `-D` handles it as a System property which has higher priority.

Regarding `pom.xml` vs `application.properties`, I believe that `application.properties` should have priority because you may not be able to change properties in the `pom` (thinking about inherited properties from a parent pom that you don't control).

Config as come a long way since the last few months, so we probably need a big refresh on the documentation. I had that on my list, but keep pushing it for later :)

**Comment by radcortez** [comment]
I believe this is already covered here:

- https://quarkus.io/guides/maven-tooling#maven-configuration-profile
- https://quarkus.io/guides/gradle-tooling#configuring-quarkus-builds
